File: examples/java/operations/dag/nested-dag.java
import software.amazon.lambda.durable.DurableContext;
import software.amazon.lambda.durable.DurableHandler;
import software.amazon.lambda.durable.dag.DagConfig;
import software.amazon.lambda.durable.dag.DagResult;

public class NestedDag extends DurableHandler<Void, Integer> {
    @Override
    public Integer handleRequest(Void input, DurableContext context) {
        DagResult result = context.dag(
                "outer",
                dag -> {
                    dag.step("prepare", String.class, (deps, ctx) -> "ready");
                    dag.dag(
                            "inner",
                            nested -> {
                                nested.step("a", Integer.class, (deps, ctx) -> 1);
                                nested.step("b", Integer.class, (deps, ctx) -> 2);
                            },
                            DagConfig.builder().maxConcurrency(5).build());
                },
                DagConfig.builder().maxConcurrency(10).build());

        return result.successCount();
    }
}

File: examples/python/operations/dag/nested-dag.py
from aws_durable_execution_sdk_python import (
    DagConfig,
    DagContext,
    DagResult,
    DurableContext,
    durable_execution,
)
@durable_execution
def handler(event: dict, context: DurableContext) -> int:
    def register(dag: DagContext) -> None:
        dag.step(lambda deps, sc: "ready", name="prepare")

        def inner(sub: DagContext) -> None:
            sub.step(lambda deps, sc: 1, name="a")
            sub.step(lambda deps, sc: 2, name="b")

        dag.dag(inner, name="inner", config=DagConfig(max_concurrency=5))

    result: DagResult = context.dag(
        register, name="outer", config=DagConfig(max_concurrency=10)
    )
    return result.success_count
